### PR TITLE
VS Code extension with code snippet templates

### PR DESCRIPTION
**Pull request by flyinghermit**
This is the first implementation of VS Code extension for Predicate. 
This PR also features [code snippets](https://code.visualstudio.com/docs/editor/userdefinedsnippets) that will help predicate policy authors with easy access to import classes as well as creating policy object and test method based on templates. 

Refer to README file for more details: [README.md](https://github.com/gravitational/predicate-lang/blob/b88baf10c8b4ce4d410cf1fe4463e12755e0bb15/plugins/vscode/README.md)

#### Demo:

[snippet-demo](https://user-images.githubusercontent.com/10177602/210734915-aac05187-071b-42c8-b78d-6fa7125b12b7.mp4)

### CODE REVIEW

File: plugins/vscode/test/helper/index.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  }
+}
+
+export async function sleep(ms: number) {
+  return new Promise(resolve => setTimeout(resolve, ms));
+}
+
+export const getDocPath = (p: string) => {
+  return path.resolve(__dirname, '../../../test/testFixture', p);
+};
```
Comment (xacrimon) [inline plugins/vscode/test/helper/index.ts:31]: inconsistent line spacing? Have we can something like prettier over this? Also, can we add prettier format checks to the CI when introducing this?

**Review by xacrimon — COMMENTED** [review]
should we track the mp4 in Git LFS instead? Usually committing larger binary assets isn't recommended with Git.

File: plugins/vscode/test/helper/index.ts (hunk 1 above)
Comment (flyinghermit) [reply plugins/vscode/test/helper/index.ts:31]: Prettier is already configured in these source files. Eslint is setup according Google typescript lint config [google gts](https://github.com/google/gts). But I will check once with our frontend team on style guide. 
Definitely, I have created this https://github.com/gravitational/predicate-lang/issues/75 to setup formatter and linter for Nodejs and python. i will create PR for that separately.

**Comment by flyinghermit** [comment]
> should we track the mp4 in Git LFS instead? Usually committing larger binary assets isn't recommended with Git.

@xacrimon, Good point on tracking in Git LFS. But I'll remove the mp4 file for now and add this again when we write docs so that we wont duplicate demo files over multiple locations.

File: plugins/vscode/test/helper/index.ts (hunk 1 above)
Comment (flyinghermit) [reply plugins/vscode/test/helper/index.ts:31]: @xacrimon I have rerun prettier and linter on all the files and I have also added new workflow that tests and runs linter on each push.

**Review by xacrimon — APPROVED** [review]

**Pull request merged**